src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -148,6 +143,29 @@ private function shouldRetry(\Throwable $e, Envelope $envelope, RetryStrategyInt
         return $retryStrategy->isRetryable($envelope, $e);
     }
 
+    private function getWaitingTime(Envelope $envelope, \Throwable $throwable, RetryStrategyInterface $retryStrategy): int
+    {
+        $delay = null;
+        if ($throwable instanceof RecoverableExceptionInterface && method_exists($throwable, 'getRetryDelay')) {
+            $delay = $throwable->getRetryDelay();
+        }
+
+        if ($throwable instanceof HandlerFailedException) {
+            $delays = [];
+            foreach ($throwable->getWrappedExceptions() as $nestedException) {
+                if ($nestedException instanceof RecoverableExceptionInterface && method_exists($nestedException, 'getRetryDelay')) {
+                    $delays[] = $nestedException->getRetryDelay();
+                }
+            }
+
+            if (\count($delays) > 0) {
+                $delay = min(array_filter($delays));
+            }
```

Comment:
```suggestion
            if ($delays = array_filter($delays)) {
                $delay = min($delays);
            }
```

Reply:
but are we sure that `0` as a delay should be filtered out?

Reply:
I've updated to not filter out zero